tests/src/com/unciv/uniques/ResourceTests.kt

```diff
@@ -0,0 +1,108 @@
+package com.unciv.uniques
+
+import com.unciv.testing.GdxTestRunner
+import org.junit.Assert
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(GdxTestRunner::class)
+class ResourceTests {
+    private val game = TestGame().apply { makeHexagonalMap(2) }
+    private val civInfo = game.addCiv()
+    private val city = game.addCity(civInfo, game.tileMap[0,0])
+
+    @Test
+    fun testConsumesResourceUnique() {
+        val consumesCoal = game.createBuilding("Consumes [1] [Coal]")
+        city.cityConstructions.addBuilding(consumesCoal.name)
+        Assert.assertTrue(civInfo.getCivResourcesByName()["Coal"] == -1)
+    }
+
+    @Test
+    fun testResourceProductionModifierDoesNotAffectConsumption() {
+        val consumesCoal = game.createBuilding("Consumes [1] [Coal]", "Double quantity of [Coal] produced")
+        city.cityConstructions.addBuilding(consumesCoal.name)
+        Assert.assertTrue(civInfo.getCivResourcesByName()["Coal"] == -1)
+    }
+
+    @Test
+    fun testResourceProductionAndConsumptionModifierDoesNotAffectConsumption() {
+        val consumesCoal = game.createBuilding("Consumes [1] [Coal]", "Provides [1] [Coal]",
+            "Double quantity of [Coal] produced")
+        city.cityConstructions.addBuilding(consumesCoal.name)
+        Assert.assertTrue(civInfo.getCivResourcesByName()["Coal"] == 1)
```

Comment:
Can this be split into at least two buildings (one consumes, one provides, the double quantity can be on either building) and at least two separate tests (one for the provide building, and one for the consume building) so it's more obvious what's going on?